pkg/custompluginmonitor/custom_plugin_monitor.go

```diff
@@ -126,7 +126,8 @@ func (c *customPluginMonitor) generateStatus(result cpmtypes.Result) *types.Stat
 			condition := &c.conditions[i]
 			if condition.Type == result.Rule.Condition {
 				status := toConditionStatus(result.ExitStatus)
-				if condition.Status != status || condition.Reason != result.Rule.Reason {
+				// change 1: Condition status change from True to False/Unknown
+				if condition.Status != status && condition.Status == types.True {
```

Comment:
`condition.Status == types.True && status != types.True`

Reply:
IIRC, `true` means there is a problem, `false` means there is no problem. If so, we should reset default here.

Reply:
> What about this? I feel like the current logic in your change is just opposite.

In your code:
status true -> false: reason is updated;
status false -> true: reason is reset to default.

Is that correct?

After a re-think, i am wrong. I will update this.